operator/podinfo/podinfo_controller.go

```diff
@@ -0,0 +1,331 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package podinfo
+
+import (
+	"context"
+
+	ciliumiov1alpha1 "github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconciler reconciles a PodInfo object
+type Reconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=cilium.io,resources=PodInfo,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=cilium.io,resources=PodInfo/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=cilium.io,resources=PodInfo/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+
+// Reconcile is the main loop of the controller that something changes with the pod resources in the cluster.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	l := log.FromContext(ctx)
+
+	// Get the Pod.
+	pod := &corev1.Pod{}
+	if err := r.Get(ctx, req.NamespacedName, pod); err != nil {
+
+		// CASE 1: Pod is deleted, therefore delete the corresponding PodInfo resource.
+		if client.IgnoreNotFound(err) == nil {
+			l.Info("Pod Deleted: ", "Name", req.Name, " Namespace", req.Namespace)
+			return r.deletePodInfo(ctx, req)
+		}
```

Comment:
curious what happens if we don't have this logic? doesn't k8s garbage collect the pod info when the owner gets deleted?

Reply:
that is a good point, Initially I planned to handle as many cases as possible and then incrementally remove/improve things.

Let me try and remove this and see if deletetion still works